src/controllers/controller.h

```diff
@@ -135,6 +133,18 @@ class Controller : public QObject {
         emit openChanged(m_bIsOpen);
     }
 
+    const QString m_sDeviceName;
+    // The QLoggingCategory constructor only takes a const char*
+    // which it does not take ownership of. So the Controller
+    // constructor needs to create temporary QStrings and convert
+    // them to QByteArrays with the same lifetime as the QLoggingCategory.
+    const QByteArray m_logBaseRawString;
```

Comment:
How about using `.send` and `.recv` instead of `.output` and `.input`? We already use the terms *send* and *receive* in C++ code.

Reply:
I don't know why but `input` and `output` feel like better terms for this... @Swiftb0y what do you think?

Reply:
Ok for me.